SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestEntityInteration/EntityInteractionTest.cpp

```diff
@@ -0,0 +1,144 @@
+#include "EntityInteractionTest.h"
+#include "EntityInteractionTestActor.h"
+#include "Kismet/GameplayStatics.h"
+
+#include "SpatialGDKSettings.h"
+
+ASpatialEntityInteractionTest::ASpatialEntityInteractionTest()
+	: Super()
+{
+	Author = "Nicolas";
+	Description = TEXT("");
+}
+
+#define EXPECT_IMMEDIATE(Actor, FunctionName, Context)                                                                                     \
+	do                                                                                                                                     \
+	{                                                                                                                                      \
+		FString DebugString("Immediate execution of a ");                                                                                  \
+		DebugString += FunctionName;                                                                                                       \
+		DebugString += " on ";                                                                                                             \
+		DebugString += Actor->GetName();                                                                                                   \
+		DebugString += " : ";                                                                                                              \
+		DebugString += Context;                                                                                                            \
+		AssertTrue(Actor->Steps.Contains(NumSteps) && Actor->Steps[NumSteps] == FunctionName, DebugString);                                \
+		++NumSteps;                                                                                                                        \
+	} while (false)
```

Comment:
nit - I'd close with ';' since this is a code block

Reply:
Actually I do that on purpose, to force the caller to use it as a function call, and append a ;
Editors tend to freak out with indentation if the ; is not added at the call site.

Reply:
However to my knowledge, all the macros in UE4 end with a semi-colon?

Reply:
They do, and that annoyed me to no end when VS messed up indentation because the semicolon was not mandatory.